<launch>
  <arg name="uav_name" default="$(optenv UAV_NAME uav2)"/>
  <arg name="mrs_id" default="$(optenv MRS_ID)"/>

  <arg name="standalone" default="false"/>

  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg uav_name)_uvdar_nodelet_manager"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>

  <arg name="threshold" default="50"/>

  <arg name="device" default="$(optenv BLUEFOX_UV)"/>
  <arg name="camera_name" default="mv_$(arg device)"/>

  <!-- <arg name="calibrations_folder" default="$(find mrs_uav_deployment)/config/uvdar_calibrations"/> -->

  <arg name="rate" default="60"/>
  <arg name="camera" default="$(arg camera_name)"/>
  <arg name="frame_id" default="$(arg camera)"/>
  <arg name="fps" default="$(arg rate)"/>
  <arg name="idpf" default="0"/>
  <arg name="aec" default="false"/>
  <arg name="expose_us" default="1000"/>
  <arg name="agc" default="false"/>
  <arg name="gain_db" default="0.0"/>
  <arg name="cbm" default="0"/>
  <arg name="ctm" default="1"/>
  <arg name="dcfm" default="0"/>
  <arg name="hdr" default="false"/>
  <arg name="wbp" default="-1"/>
  <arg name="request" default="0"/>
  <arg name="mm" default="0"/>
  <arg name="jpeg_quality" default="90"/>

  <arg name="blink_process_rate" default="10"/>

  <arg name="accumulator_length"         default="12"/>
  <arg name="pitch_steps"         default="16"/>
  <arg name="yaw_steps"         default="16"/>
  <arg name="max_pixel_shift"         default="4"/>

  <arg name="filterDistLength"         default="10"/>
  <arg name="filterOrientationLength"         default="10"/>

  <arg name="debug" default="false"/>
  <arg name="visual_debug" default="false"/>
  <arg name="gui" default="false"/>
  <arg name="publish_visualization" default="true"/>
  <arg name="visualization_rate" default="2"/>
  <!-- <arg name="gui" default="false"/> -->

  <node
    name="uvcam_left_tf_$(arg uav_name)"
    pkg="tf"
    type="static_transform_publisher"
    args="0.04  0.11 0.06 -0.3490658504 0.0 -1.57079632679 $(arg uav_name)/fcu $(arg uav_name)/uvcam_left 100"/>

  <!-- ideal angle: 1.221730476 - pi/2 -->

      <!-- <origin xyz="0.108 0 0.085" rpy="1.57079632679 4.71238898038 1.57079632679" /> -->
  <!--Usage: static_transform_publisher x y z yaw pitch roll frame_id child_frame_id  period(milliseconds)-->
<!-- OR -->
  <group ns="$(arg uav_name)">
  
    <!-- <node pkg="nodelet" type="nodelet" name="$(arg uav_name)_uvdar_nodelet_manager" args="manager" output="screen" launch-prefix="debug_roslaunch"> -->
    <node pkg="nodelet" type="nodelet" name="$(arg uav_name)_uvdar_nodelet_manager" args="manager" output="screen" >
      <param name="num_worker_threads" value="8" />
    </node>

    <node pkg="nodelet" type="nodelet" name="uvdar_bluefox" args="$(arg nodelet) bluefox2/SingleNodelet $(arg nodelet_manager)" respawn="true" output="screen" launch-prefix="bash -c 'sleep 5; $0 $@'">
      <param name="camera_name" type="string" value="$(arg camera_name)"/>
      <param name="identifier" type="string" value="$(arg device)"/>

      <param name="frame_id" type="string" value="$(arg frame_id)"/>
      <param name="fps" type="double" value="$(arg fps)"/>
      <param name="idpf" type="int" value="$(arg idpf)"/>
      <param name="aec" type="bool" value="$(arg aec)"/>
      <param name="expose_us" type="int" value="$(arg expose_us)"/>
      <param name="agc" type="bool" value="$(arg agc)"/>
      <param name="gain_db" type="double" value="$(arg gain_db)"/>
      <param name="cbm" type="int" value="$(arg cbm)"/>
      <param name="ctm" type="int" value="$(arg ctm)"/>
      <param name="dcfm" type="int" value="$(arg dcfm)"/>
      <param name="hdr" type="bool" value="$(arg hdr)"/>
      <param name="wbp" type="int" value="$(arg wbp)"/>
      <param name="request" type="int" value="$(arg request)"/>
      <param name="mm" type="int" value="$(arg mm)"/>

      <param name="image_raw/compressed/jpeg_quality" type="int" value="$(arg jpeg_quality)"/>
      <param name="image_raw/theora/keyframe_frequency" value="60" />
      <param name="image_raw/theora/target_bitrate" value="50000" />
      <param name="image_raw/theora/quality" value="8" />
      <param name="image_raw/theora/optimize_for" value="0" />

      <remap from="/$(arg uav_name)/uvdar_bluefox/image_raw" to="/$(arg uav_name)/uvdar_bluefox/left/image_raw"/>
    </node>




  </group>


  <!-- </group> -->
</launch>
